src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs

```diff
@@ -526,8 +526,7 @@ protected virtual void ExecuteEvmCall(
 
                 if (substate.ShouldRevert || substate.IsError)
                 {
-                    if (Logger.IsTrace)
-                        Logger.Trace("Restoring state from before transaction");
+                    if (Logger.IsTrace) Logger.Trace("Restoring state from before transaction");
```

Comment:
`goto Fail;` here too?

Reply:
You get redund on revert would cause it to skip 

`spentGas = Refund...`